### PR TITLE
Update and align Maven coordinates

### LINKED ISSUE #1931: Correct the maven coordinates in version 1.0
The current coordinate of `ktlint-cli` is `com.pinterest:ktlint:0.48.2`, we should move it to `com.pinterest.ktlint:ktlint-cli:1.0` in 1.0, more context in #1929.

Next to this the artifact names of modules below have to be fixed:

| Current name | New name |
| :-- | :-- |
| ktlint-cli-reporter | ktlint-cli-reporter-core |
| ktlint-reporter-baseline | ktlint-cli-reporter-baseline |
| ktlint-reporter-checkstyle | ktlint-cli-reporter-checkstyle |
| ktlint-reporter-format | ktlint-cli-reporter-format |
| ktlint-reporter-json | ktlint-cli-reporter-json |
| ktlint-reporter-plain | ktlint-cli-reporter-plain |
| ktlint-reporter-plain-summary | ktlint-cli-reporter-plain-summary |
| ktlint-reporter-sarif | ktlint-cli-reporter-sarif |

### PR DESCRIPTION
**Pull request by paul-dingemans**
## Description

Closes #1931

## Checklist

Before submitting the PR, please check following (checks which are not relevant may be ignored):
- [X] Commit message are well written. In addition to a short title, the commit message also explain why a change is made.
- [X] At least one commit message contains a reference `Closes #<xxx>` or `Fixes #<xxx>` (replace`<xxx>` with issue number)
- [ ] Tests are added
- [ ] KtLint format has been applied on source code itself and violations are fixed
- [X] `CHANGELOG.md` is updated
- [X] PR description added

[Documentation](https://pinterest.github.io/ktlint/) is updated. See [difference between snapshot and release documentation](https://github.com/pinterest/ktlint/tree/master/documentation)
- [X] [Snapshot documentation](https://github.com/pinterest/ktlint/tree/master/documentation/snapshot) in case documentation is to be released together with a code change
- [ ] [Release documentation](https://github.com/pinterest/ktlint/tree/master/documentation/release-latest) in case documentation is related to a released version of ktlint and has to be published as soon as the change is merged to master 
- [ ] In case of adding a new rule, it needs to be added to [experimental rules documentation](https://pinterest.github.io/ktlint/latest/rules/experimental/)

### CODE REVIEW

File: build-logic/src/main/kotlin/ktlint-publication.gradle.kts [hunk 1]
```diff
@@ -15,14 +15,11 @@ project.version =
     } else {
         definedVersion
     }
-// TODO: Remove `localGradleProperty` once https://github.com/gradle/gradle/issues/23572 or https://github.com/pinterest/ktlint/issues/1931 is fixed.
-project.group = localGradleProperty("GROUP").orNull ?: providers.gradleProperty("GROUP").orNull
-    ?: throw GradleException("Project group property is missing")
```
Comment (Goooler) [inline build-logic/src/main/kotlin/ktlint-publication.gradle.kts:20]: We can still use `project.group = providers.gradleProperty("GROUP").get()` here, to align with `project.version`.

File: build-logic/src/main/kotlin/ktlint-publication.gradle.kts [hunk 2]
```diff
@@ -15,14 +15,11 @@ project.version =
...
 
 publishing {
     publications {
         create<MavenPublication>("maven") {
-            groupId = group.toString()
+            groupId = "com.pinterest.ktlint"
```
Comment (Goooler) [inline build-logic/src/main/kotlin/ktlint-publication.gradle.kts:22]: And revert this line.

**Review by Goooler — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,23 @@ This project adheres to [Semantic Versioning](https://semver.org/).
 
 ## Unreleased
 
+IMPORTANT: Maven coordinates have been changed for some module. Be sure to update them, to get latest changes!
+
+| Old Maven coordinates                              | New Maven coordinates                                  |
+|----------------------------------------------------|--------------------------------------------------------|
+| com.pinterest.ktlint                               | com.pinterest.ktlint.ktlint-cli                        |
```
Comment (PaulWoitaschek) [inline CHANGELOG.md:11]: I think the coordinates are actually with a double colon:

I.e.
```com.pinterest:ktlint` -> `com.pinterest.ktlint:ktlint-cli`

Comment (paul-dingemans) [reply CHANGELOG.md:11]: Tnx. Fixed via #2264

**Pull request merged**